File: ManLuxe_frontend/src/app/models/product.ts
export interface ProductSize {
  id: number;
  size: string;
  stock: number;
}

export interface Product {

  id: number;
  name: string;
  description: string;
  price: number;
  discount: number;
  stock: number;
  brand: string;
  gender: string;
  imageUrl: string;
  active: boolean;
  categoryId: number;
  categoryName: string;

  sizeType: string;

  sizes: ProductSize[];
}

export interface ProductRequest {

  name: string;
  description: string;
  price: number;
  discount: number;
  stock: number;
  brand: string;
  gender: string;
  imageUrl: string;
  active: boolean;
  categoryId: number;

  sizeType: string;

  sizes: {
    size: string;
    stock: number;
  }[];
}

File: FirstCopy_backend/src/main/java/com/example/FirstCopy/entity/Product.java
package com.example.FirstCopy.entity;
import java.util.ArrayList;
import java.util.List;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import com.example.FirstCopy.entity.Gender;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "products")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(length = 1000)
    private String description;

    @Column(nullable = false)
    private Double price;

    private Double discount = 0.0;

    @Column(nullable = false)
    private Integer stock;


    private String brand;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Gender gender;

    private String imageUrl;

    private Boolean active = true;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    @JsonIgnoreProperties({"products"})
    private Category category;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    @OneToMany(
            mappedBy = "product",
            cascade = CascadeType.ALL,
            orphanRemoval = true
    )
    private List<ProductSize> sizes = new ArrayList<>();

    @PrePersist
    public void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private SizeType sizeType;

    @PreUpdate
    public void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}